evap/staff/views.py

```diff
@@ -148,7 +148,10 @@ def semester_course_operation(request, semester_id):
         raise SuspiciousOperation("Unknown target state: " + target_state)
 
     course_ids = (request.GET if request.method == 'GET' else request.POST).getlist('course')
-    courses = Course.objects.filter(id__in=course_ids)
+    courses = Course.objects.filter(id__in=course_ids).annotate(
+        midterm_grade_documents_count=Count("grade_documents", filter=Q(grade_documents__type=GradeDocument.MIDTERM_GRADES), distinct=True),
+        final_grade_documents_count=Count("grade_documents", filter=Q(grade_documents__type=GradeDocument.FINAL_GRADES), distinct=True)
+    )
```

Comment:
This small change also displays the counts in the UI? Awesome :D. Sometimes I wish I could use django's easy way of doing things in other web applications...

Reply:
well, we already had code for displaying these counts in the template (in an included template to be precise). to me it looks like it previously had some icon plus an empty span.

@janno42, in that case shouldn't the commit message rather be "fix" instead of "add"?

Reply:
yes, it's a fix. will change the message.

Reply:
Ah well.